lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -77,6 +94,50 @@ import qualified Data.ByteArray as BA
 import qualified Data.ByteString as BS
 import qualified Data.ByteString.Lazy as BL
 
+{-------------------------------------------------------------------------------
+                                   Key Types
+-------------------------------------------------------------------------------}
+
+-- | Data for generating random scheme addresses
+data RndKey (level :: Depth) key = RndKey
+    { masterKey :: Key 'RootK key
+    -- ^ The root key, generated from the seed.
+    , accountKey :: Key 'AccountK key
+    -- ^ The account key, generated from the root key and account index.
+    , addressKey :: Key 'AddressK key
+    -- ^ The address key, generated from the account key and address index.
+    , accountIndex :: Index 'Hardened 'AccountK
+    -- ^ Which account in the hierarchical scheme to generate addresses in.
+    , addressIndex :: Index 'Soft 'AddressK
+    -- ^ The index of the address to generate (will be randomly chosen).
+    , payloadPassphrase :: Passphrase "address-der"
+    -- ^ Used for encryption of payload containing address derivation path.
+    } deriving stock (Generic, Show, Eq)
+
+-- | A single key, with phantom types to prevent using them for the wrong
+-- purpose.
+newtype Key (level :: Depth) key = Key { getKey :: key }
+    deriving stock (Generic, Show, Eq, Functor)
+
+instance (NFData key) => NFData (Key level key)
+
+instance WalletKey RndKey where
+    type WalletKeySeed RndKey = Passphrase "seed"
+    unsafeGenerateKeyFromSeed = unsafeGenerateKeyFromSeedRnd
```

Comment:
I pulled the generate functions out of the class. But actually, whether to put this in the type class might depend on how we generalise the `Server.postWallet` function.

Reply:
@rvl We'll have different endpoints for creating wallets using sequential and random scheme, because they have fundamentally different parameters. 
So, in practice, this is where we make the concrete choice for the key type (and wallet layer).